src/dotnet/commands/dotnet-publish/PublishCommand.cs

```diff
@@ -272,33 +316,60 @@ private static void PublishRefs(LibraryExport export, string outputPath, bool de
             }
         }
 
-        private static int PublishHost(ProjectContext context, string outputPath, CommonCompilerOptions compilationOptions)
+        private static int RenamePublishedHost(ProjectContext context, string outputPath, CommonCompilerOptions compilationOptions)
         {
             if (context.TargetFramework.IsDesktop())
             {
                 return 0;
             }
 
-            foreach (var binaryName in Constants.HostBinaryNames)
+            var publishedHostFile = ResolvePublishedHostFile(outputPath);
+            if (publishedHostFile == null)
             {
-                var hostBinaryPath = Path.Combine(AppContext.BaseDirectory, binaryName);
-                if (!File.Exists(hostBinaryPath))
-                {
-                    Reporter.Error.WriteLine($"Cannot find {binaryName} in the dotnet directory.".Red());
-                    return 1;
-                }
+                Reporter.Output.WriteLine($"publish: warning: host executable not available in dependencies, using host for current platform");
```

Comment:
The primary reason I relied on old behavior was to temporarily limit just how breaking this change is (even in our own repo). 

I'm working on the changes in our own repo so we can remove the fallback behavior (it's a decent amount).

I do agree that we should remove this as it is completely inconsistent.